Add anvil-rstudio-bioconductor image

nturaga opened the pull request:
This pull request adds the anvil-rstudio-bioconductor image using the anvil-rstudio-base image to inherit from.

1. There are a couple of fixes on the anvil-rstudio-base images, most importantly fixing the rocker/tidyverse:3.6.1 tag, and then using BiocManager to install packages.

1. It also updates the README files of the repository, anvil-rstudio-base, and anvil-rstudio-bioconductor. 

Not entirely sure, if the testing framework in Leonardo needs to be complete before this is added. I'll leave it up to the reviewer's discretion.

nturaga committed:
Fix rocker/tidyverse:3.6.1 as version

Change installation of packages with BiocManager as tidyverse takes care of installing BiocManager.

Inline comments, to change location where R-packages are installed

nturaga committed:
Add initial anvil-rstudio-bioconductor image,

- Make changes to version and CHANGELOG in anvil-rstudio-base image

nturaga committed:
Move xvfb_init to deps/ folder

nturaga committed:
Update README files in all places

nturaga commented:
This PR is waiting for a reviewer and also a "Test" environment for the terra platform @rtitle @Qi77Qi .

I'll try to look into this image more and make any changes as needed.

rtitle commented:
Nice! I'll look this over and try it out.

rtitle commented on anvil-rstudio-bioconductor/Dockerfile:
```diff
@@ -0,0 +1,124 @@
+FROM anvil-rstudio-base:0.0.2
```
This needs to be
```
FROM us.gcr.io/anvil-gcr-public/anvil-rstudio-base:0.0.2
```
(I got a "repository not found" error otherwise)

nturaga replied:
Yes, so, I didn't push anything up to us.gcr.io/anvil-gcr-public because I couldn't actually launch these images on the terra platform to see and RStudio image. 

But I can push an image to `us.gcr.io/anvil-gcr-public/anvil-rstudio-base:0.0.2` so we can run it this way.

nturaga replied:
Nvm, It seems you just pushed to the anvil-gcr-public directory. Thank you :D

rtitle replied:
Yep :)

rtitle commented:
I pushed both images to https://console.cloud.google.com/gcr/images/anvil-gcr-public?project=anvil-gcr-public. I then launched the bioconductor one in a Leo test environment. It seems to work -- I didn't do anything too fancy, but I was able to launch RStudio and run some commands.

@nturaga feel free to merge this once the above comment is resolved.

rtitle reviewed (approved):
see above

nturaga committed:
Update base image name from GCR

- Add Bioconductor packages to be installed on the RStudio image. This
keeps it consistent with the Jupyter-bioconductor image.
